zerver/views/streams.py

```diff
@@ -67,7 +68,7 @@
     filter_stream_authorization,
     list_to_streams,
 )
-from zerver.lib.topic import get_topic_history_for_stream, messages_for_topic
+from zerver.lib.topic import get_topic_history_for_stream, messages_for_topic, get_topic_history_for_web_public_stream
```

Comment:
I think we could just pass `public_history=True` to `get_topic_history_for_stream` and delete `get_topic_history_for_web_public_stream`; it looks like that does the same thing.

Or maybe for type-checking reasons (i.e. avoiding adding an `Optional[UserProfile]`) it's better to do it the other way around, by having `get_topic_history_for_stream` just call `get_topic_history_for_public_stream` (renamed from `get_topic_history_for_web_public_stream`) in its `else` clause.